smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/HttpBindingProtocolGenerator.kt

```diff
@@ -216,50 +212,73 @@ abstract class HttpBindingProtocolGenerator : ProtocolGenerator {
         }
     }
 
-    private fun renderCodableExtension(ctx: ProtocolGenerator.GenerationContext, shape: Shape) {
+    fun renderCodableExtension(
+        ctx: ProtocolGenerator.GenerationContext,
+        shape: Shape,
+        encodable: Boolean = true,
+        decodable: Boolean = true
+    ) {
```

Comment:
Selectively render `Encodable`, `Decodable`, or `Codable` since model types may not implement both encoding & decoding anymore.